components/script/script_thread.rs

```diff
@@ -1530,6 +1530,14 @@ impl ScriptThread {
             self.perform_a_microtask_checkpoint(can_gc);
         }
 
+        for (_, doc) in self.documents.borrow().iter() {
+            let window = doc.window();
+            let _realm = enter_realm(&*doc);
```

Comment:
I don't think the checkpoint does anything requiring a realm, so this can be removed.